### ISSUE TITLE
Cannot use precreated Config DB with SPFarm

### ISSUE THREAD

**Issue by aboersch** [PowerShell/SharePointDsc#574]
**Details of the scenario you try and problem that is occurring:**
I am trying to create a new farm with a pre-created empty database. This used to work with SPCreateFarm.

There is a compatibility range mismatch between the Web server and database "", and connections to the data have been blocked to due to this incompatibility. This can happen when a content database has not been upgraded to be within the compatibility range of the Web server, or if the database has been upgraded to a higher level than the web server. The Web server and the database must be upgraded to the same version and build level to return to compatibility range.
    + CategoryInfo          : InvalidData: (Microsoft.Share...urationDatabase:) [], CimException
    + FullyQualifiedErrorId : Microsoft.SharePoint.PowerShell.SPCmdletConnectSPConfigurationDatabase
    + PSComputerName        : localhost

Is this perhaps by design? Or was I supposed to create the new _Lock Database as well?

**The DSC configuration that is using the resource:**

**Version of the Operating System and PowerShell the DSC Target Node is running:**
Windows Server 2016
SP2016
SQL2016

**Version of the DSC module you're using:**
1.6

**Comment by aboersch** [comment]
Or is this perhaps due to SP2016. I am seeing errors for usage and state service apps.

```
The specified database already exists.
    + CategoryInfo          : InvalidData: (Microsoft.Offic...eDatabaseCmdlet:) [], CimException
    + FullyQualifiedErrorId : Microsoft.Office.Server.Administration.NewStateServiceDatabaseCmdlet
    + PSComputerName        : localhost
```

**Comment by BrianFarnhill** [comment]
OK so the _Lock database is literally just a flag to our logic to say that the farm is in the process of being created, we delete it once the farm is established so its never there long term.

This error tells me that it's not happy with the pre-created database. Are you able to provide me some details on how you created the database so I can try to look in to this?

**Comment by aboersch** [comment]
``` powershell
param(
    $SQLServer,
    $SQLInstance,
    $DBName
)

function ConnectSql([string] $SQLServer = $env:COMPUTERNAME,[string]$SQLInstanceName = "MSSQLSERVER")
{
    [System.Reflection.Assembly]::LoadWithPartialName('Microsoft.SqlServer.Smo') | Out-Null
    if($SQLInstanceName -eq "MSSQLSERVER")
    {
        $ConnectSQL = $SQLServer
    }
    else
    {
        $ConnectSQL = "$SQLServer\$SQLInstanceName"
    }

    Write-Verbose "Connecting to SQL $ConnectSQL"
    $SQL = New-Object Microsoft.SqlServer.Management.Smo.Server $ConnectSQL

    if($SQL)
    {
        Write-Verbose "Connected to SQL $ConnectSQL"
    }
    else
    {
        throw "Failed connecting to SQL $ConnectSQL"
    }
    return $SQL
}

$sql = ConnectSql $SQLServer $SQLServerInstance

$db = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Database -ArgumentList $sql,$DBname
$db.Create()

$additionalOwners = @("DOMAIN\SPSetup","DOMAIN\SPFarm")
foreach($additionalOwner in $this.IncludeAdditionalOwners) {
    if (!($db.Users.Contains($additionalOwner)))
    {
        $user = New-Object Microsoft.SqlServer.Management.Smo.User($db,[string]$additionalOwner)
        $user.Login = $additionalOwner
        $user.Create()
    } else {
        $user = $db.Users[$additionalOwner]
    }
    $user.AddToRole("db_owner")
}

```

**Comment by ykuijs** [comment]
Haven't done any troubleshooting yet, but does your SQL installation have the correct collation, Latin1_General_CI_AS_KS_WS (which is Case Insensitive, Accent Sensitive, Kana Sensitive and Width Sensitive)?

If not, your precreated database probably is created with the incorrect collation and that probably is the reason why SharePoint isn't accepting it. If SharePoint creates your databases, it makes sure the databases are created with the collation required by SharePoint. Your script doesn't specify a collation, so the default collation is used.

**Comment by bevanm8** [comment]
I am having the same issue in 2 different environments. SharePoint 2016 and SQL Server 2014, Server 2012 R2. Only difference here is I used SharePointDSC to build both of them. 
Unfortunately the build in both environments is stuck on the same error as above. Seems to be erroring somewhere in the SPFarm module (See output attached), the SP_Config database gets created, but the Admin Content database does not. I've checked all the SQL permissions etc, tried upgrading the database, nothing seems to work. I feel like I must be missing a basic step for this to happen in both environments. I find it interesting that the error has nothing after database between the quotes e.g "There is a compatibility range mismatch between the Web server and **database ""**, and connections to the data have been blocked to due to this incompatibility. "
[SharePoint DSC Errors.txt](https://github.com/PowerShell/SharePointDsc/files/1079886/SharePoint.DSC.Errors.txt)
